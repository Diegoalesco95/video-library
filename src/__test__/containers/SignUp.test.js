import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import ProviderMock from '../../__mocks__/ProviderMock';
import SignUp from '../../containers/Signup';

describe('SignUp Container', () => {
  test('Match Snapshot', () => {
    const signup = create(
      <ProviderMock>
        <SignUp />
      </ProviderMock>,
    );
    expect(signup.toJSON()).toMatchSnapshot();
  });

  it('Calls and executes preventDefault function onSubmit form', () => {
    const preventDefault = jest.fn();
    const wrapper = mount(
      <ProviderMock>
        <SignUp />
      </ProviderMock>,
    );
    wrapper.find('form').simulate('submit', { preventDefault });
    expect(preventDefault).toHaveBeenCalledTimes(1);
    wrapper.unmount();
  });
});
