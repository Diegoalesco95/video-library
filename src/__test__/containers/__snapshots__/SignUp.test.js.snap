// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Container Match Snapshot 1`] = `
Array [
  <header
    className="header isSignUp"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo Platzi Video"
        className="header__img"
        src="test-file.stub"
      />
    </a>
    <div
      className="header__menu"
    >
      <div
        className="header__menu--profile"
      >
        <img
          alt="user Icon"
          src="test-file.stub"
        />
        <p>
          Menu
        </p>
      </div>
      <ul>
        <li>
          <a
            href="/login"
            onClick={[Function]}
          >
            Iniciar Sesión
          </a>
        </li>
        <li>
          <a
            href="/signUp"
            onClick={[Function]}
          >
            Registrarse
          </a>
        </li>
      </ul>
    </div>
  </header>,
  <section
    className="register"
  >
    <section
      className="register__container"
    >
      <h2>
        Regístrate
      </h2>
      <form
        className="register__container--form"
        onSubmit={[Function]}
      >
        <input
          aria-label="Nombre"
          className="input--signup"
          id="name"
          name="name"
          onChange={[Function]}
          placeholder="Nombre"
          type="text"
        />
        <input
          aria-label="Correo"
          className="input--signup"
          id="email"
          name="email"
          onChange={[Function]}
          placeholder="Correo"
          type="email"
        />
        <input
          aria-label="Contraseña"
          className="input--signup"
          id="password"
          name="password"
          onChange={[Function]}
          placeholder="Contraseña"
          type="password"
        />
        <input
          aria-label="Confirmar contraseña"
          className="input--signup"
          id="confirm-password"
          name="passwordConfirmation"
          placeholder="Confirmar contraseña"
          type="password"
        />
        <button
          className="button"
          type="submit"
        >
          Registrarme
        </button>
      </form>
      <section
        className="register__container--social-media"
      >
        <div
          className="google"
        >
          <img
            alt="GoogleIcon"
            className="google__img"
            src="test-file.stub"
          />
          <p>
            <a
              href="/auth/google/"
            >
              Registro con Google 
            </a>
          </p>
        </div>
        <div
          className="twitter"
        >
          <img
            alt="TwitterIcon"
            className="twitter__img"
            src="test-file.stub"
          />
          <p>
            <a
              href="/auth/twitter"
            >
              Registro con Twitter
            </a>
          </p>
        </div>
        <div
          className="facebook"
        >
          <img
            alt="FacebookIcon"
            className="facebook__img"
            src="test-file.stub"
          />
          <p>
            <a
              href="/auth/facebook"
            >
              Registro con Facebook
            </a>
          </p>
        </div>
      </section>
      <p
        className="register__container--login"
      >
        ¿Ya tienes una cuenta?
        <br />
        <a
          href="/login"
          onClick={[Function]}
        >
          Inicia sesión
        </a>
      </p>
    </section>
  </section>,
]
`;
